### DISCUSSION TITLE
Feature request: pitch shifter

### DISCUSSION THREAD

**Discussion by 0joshuaolson1** [kfrlib/kfr#214] [Ideas]
One question is how to do this with arbitrarily good quality (FIR?, alternatives to overlap/add?).

**Comment by dancasarin** [comment]
There are various algorithms that can give good quality, but choice highly depends on a type of input audio you want to process.
For instance, SOLA, phase vocoder, Sinusoidal spectral modeling, neural networks.
Many of pitch shift algorithms are based on FFT which is already in KFR.
For some of these algorithms, sample rate conversion must be applied after processing to return the signal its original sample rate.
Vocal pitch shift that preserves formant is more complex and requires additional processing.
